integration-tests/src/test/java/com/datastax/dse/driver/api/core/graph/remote/GraphTraversalRemoteITBase.java

```diff
@@ -643,9 +644,9 @@ public void should_allow_use_of_dsl_graph_binary() {
    */
   @Test
   public void should_return_correct_results_when_bulked() {
-    Optional<Version> dseVersion = ccmRule().getCcmBridge().getDseVersion();
     Assumptions.assumeThat(
-            dseVersion.isPresent() && dseVersion.get().compareTo(Version.parse("5.1.2")) > 0)
+            CcmBridge.isDistributionOf(BackendType.DSE)
+                && CcmBridge.getDistributionVersion().compareTo(Version.parse("5.1.2")) > 0)
```

Comment:
This feels like we're just moving the problem around a bit.  We haven't decreased the verbosity... we've just moved it around a bit.

There are a lot of instances of things that use this pattern:

```
Are we DSE/OSS/HCD && are we greater than/less than/something else some specific version
```

The removal of the Optional types here (i.e. the removal of CcmBridge.getDseVersion() below) means that we have to remember to do both of the checks above manually.  If we were to make better use of the Optional type, however, we could do the whole thing in a single check.  For example, the statements above can also be expressed as:

```
Assumptions.assumeThat(
    ccmRule().getCcmBridge().getDseVersion().map(v -> v.compareTo(Version.parse("5.1.2")) > 0).orElse(false))
    .isTrue();
```

In this case if getDseVersion() returns an empty Optional (because we aren't dealing with DSE) the orElse() will return false.  Otherwise we do the integer compare and return true or false based on it's results.  This approach allows us to encapsulate all the checks we want into a single line of logic.  I'd argue that's more desirable in a test framework like this where you want to make sure your base assumptions are respected before moving on to the actual tests.